tpm2/tpm2.go

```diff
@@ -353,25 +353,47 @@ func encodeCreateRawTemplate(owner tpmutil.Handle, sel PCRSelection, parentPassw
 	)
 }
 
-func decodeCreatePrimary(in []byte) (tpmutil.Handle, crypto.PublicKey, error) {
-	var handle tpmutil.Handle
+func decodeCreatePrimary(in []byte) (handle tpmutil.Handle, public, creationData, creationHash []byte, ticket *Ticket, creationName []byte, err error) {
 	var paramSize uint32
 
 	buf := bytes.NewBuffer(in)
 	// Handle and auth data.
 	if err := tpmutil.UnpackBuf(buf, &handle, &paramSize); err != nil {
-		return 0, nil, err
+		return 0, nil, nil, nil, nil, nil, fmt.Errorf("decoding handle, paramSize: %v", err)
 	}
 
-	var public []byte
-	if err := tpmutil.UnpackBuf(buf, &public); err != nil {
-		return 0, nil, fmt.Errorf("decoding TPM2B_PUBLIC: %v", err)
+	if err := tpmutil.UnpackBuf(buf, &public, &creationData, &creationHash); err != nil {
+		return 0, nil, nil, nil, nil, nil, fmt.Errorf("decoding public, creationData, creationHash: %v", err)
 	}
 
-	pub, err := DecodePublic(public)
+	ticket, err = decodeTicket(buf)
+	if err != nil {
+		return 0, nil, nil, nil, nil, nil, fmt.Errorf("decoding ticket: %v", err)
+	}
+
+	if err := tpmutil.UnpackBuf(buf, &creationName); err != nil {
+		return 0, nil, nil, nil, nil, nil, fmt.Errorf("decoding creationName: %v", err)
+	}
+
+	if _, err := DecodeCreationData(creationData); err != nil {
+		return 0, nil, nil, nil, nil, nil, fmt.Errorf("parsing CreationData: %v", err)
+	}
+	return
```

Comment:
Explicitly spell out all return vars, don't do naked returns. It can lead to confusing behavior.